### ISSUE TITLE
oauth2 with nextcloud

### ISSUE THREAD

**Issue by markuman** [Place1/wg-access-server#72]
https://docs.nextcloud.com/server/19/developer_manual/client_apis/LoginFlow/index.html#login-flow-v2

I try to configure the oidc (which base on oauth2) against a nextcloud instance.  
When I use as `issuer`value `https://cloud.example.com/index.php/login/v2`, I got an error  

`FATA[0000]oidc.go:39 failed to create oidc provider: oidc: failed to decode provider discovery object: expected Content-Type = application/json, got "text/html; charset=UTF-8": invalid character '<' looking for beginning of value `

I'm a bit lost. Or is plain oauth2 not supported? Any chance to support it too?

**Comment by Place1** [comment]
Yeah the OIDC provider doesn't support plain old oauth2. We can open a feature request issue for oauth2 and I or perhaps someone from the community could help out adding it. Support for oauth2 should be straight forward because the current OIDC provider can be adapted for it.

**Comment by markuman** [comment]
> We can open a feature request issue for oauth2 and I or perhaps someone from the community could help out adding it. 

That would be awesome!
